sts/client_test.go

```diff
@@ -380,3 +386,34 @@ func TestIssueActAs(t *testing.T) {
 
 	t.Logf("expires in %s", s.Lifetime.Expires.Sub(s.Lifetime.Created))
 }
+
+func TestNewClient(t *testing.T) {
+	t.Run("Happy path client creation", func(t *testing.T) {
+		simulator.Test(func(ctx context.Context, client *vim25.Client) {
+			_, err := NewClient(ctx, client)
+			require.NoError(t, err)
+		})
+	})
+
+	t.Run("STS client should work with Envoy sidecar even when lookup service is down", func(t *testing.T) {
+		model := simulator.VPX()
+
+		model.Create()
```

Comment:
Does this test need model.Create() to happen here? Suppose it doesn't hurt, but looks like default simulator.Test behavior would work the same here.

Reply:
Hmm, yeah, good point. Might have been leftover from me trying to fiddle with the TLS config before I realized it was the `new(tls.Config)` thing tripping me up

Reply:
Ah, I remember. The existing Test/Run helpers, if not given a Model, will create a blank one with a Nil service. In that path, the TLS config is always set up (ie. simulator serves HTTPS) - which breaks the UT, as that requires HTTP

Reply:
I suppose I could in theory change the Test/Run helpers but that seemed like a larger blast radius.

Reply:
Ah gotcha. Haven't tried, but seems like that'd need the patch below? 

```diff
modified   simulator/model.go
@@ -876,9 +876,10 @@ func (m *Model) Run(f func(context.Context, *vim25.Client) error) error {
 		if err != nil {
 			return err
 		}
+	} else {
+		m.Service.TLS = new(tls.Config)
 	}
 
-	m.Service.TLS = new(tls.Config)
 	m.Service.RegisterEndpoints = true
 
 	s := m.Service.NewServer()
```